cmd/kubeadm/app/phases/addons/dns/dns.go

```diff
@@ -371,31 +371,18 @@ func IsCoreDNSConfigMapMigrationRequired(corefile string) bool {
 }
 
 func migrateCoreDNSCorefile(client clientset.Interface, cm *v1.ConfigMap, corefile, currentInstalledCoreDNSVersion string) error {
-	// Take a copy of the Corefile data as `Corefile-backup` and update the ConfigMap
-	// Also point the CoreDNS deployment to the `Corefile-backup` data.
-
-	if _, err := client.CoreV1().ConfigMaps(cm.ObjectMeta.Namespace).Update(context.TODO(), &v1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      kubeadmconstants.CoreDNSConfigMap,
-			Namespace: metav1.NamespaceSystem,
-		},
-		Data: map[string]string{
-			"Corefile":        corefile,
-			"Corefile-backup": corefile,
-		},
-	}, metav1.UpdateOptions{}); err != nil {
-		return errors.Wrap(err, "unable to update the CoreDNS ConfigMap with backup Corefile")
-	}
-	if err := patchCoreDNSDeployment(client, "Corefile-backup"); err != nil {
-		return err
-	}
-
 	// Since the current configuration present is not the default version, try and migrate it.
 	updatedCorefile, err := migration.Migrate(currentInstalledCoreDNSVersion, kubeadmconstants.CoreDNSVersion, corefile, false)
 	if err != nil {
 		return errors.Wrap(err, "unable to migrate CoreDNS ConfigMap")
 	}
 
+	// Point the CoreDNS deployment to the `Corefile-backup` data.
+	if err := patchCoreDNSDeployment(client, "Corefile-backup"); err != nil {
```

Comment:
This step points CoreDNS Deployment to a Configmap value that doesn't exist until the next step (or exists from a prior kubeadm upgrade). IMO, we should create the backup first, then point the deployment to it.

Minor nit: the name of the function `patchCoreDNSDeployment()` seems too generic for what it does.  I think `setCorefile()` or something like that would be better.